### ISSUE TITLE
Unit tests fail on PHP 8.1

### ISSUE THREAD

**Issue by schmengler** [hyva-themes/magento2-graphql-view-model#5]
**Module version:** 1.0.3
**PHP version:** 8.1.10
**Magento version:** 2.4.5-p1
`webonyx/graphql-php` **version:** 14.11.8 (via `magento/framework`, not required by this module)

### Full output:

```
PHPUnit 9.5.27 by Sebastian Bergmann and contributors.

...EEE........                                                    14 / 14 (100%)

Time: 00:00.072, Memory: 10.00 MB

There were 3 errors:

1) Hyva\GraphqlViewModel\Model\GraphqlQueryEditorTest::testSetsGivenArgumentsOnQuery with data set "no prefix" ('', '')
TypeError: GraphQL\Language\Printer::GraphQL\Language\{closure}(): Return value must be of type string, int returned

/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Printer.php:194
/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Visitor.php:277
/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Printer.php:463
/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Printer.php:90
/var/www/html/src/vendor/hyva-themes/magento2-graphql-view-model/src/Model/GraphqlQueryEditor.php:98
/var/www/html/src/vendor/hyva-themes/magento2-graphql-view-model/tests/unit/Model/GraphqlQueryEditorTest.php:119

2) Hyva\GraphqlViewModel\Model\GraphqlQueryEditorTest::testSetsGivenArgumentsOnQuery with data set "only query" ('query ', '')
TypeError: GraphQL\Language\Printer::GraphQL\Language\{closure}(): Return value must be of type string, int returned

/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Printer.php:194
/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Visitor.php:277
/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Printer.php:463
/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Printer.php:90
/var/www/html/src/vendor/hyva-themes/magento2-graphql-view-model/src/Model/GraphqlQueryEditor.php:98
/var/www/html/src/vendor/hyva-themes/magento2-graphql-view-model/tests/unit/Model/GraphqlQueryEditorTest.php:119

3) Hyva\GraphqlViewModel\Model\GraphqlQueryEditorTest::testSetsGivenArgumentsOnQuery with data set "named query" ('query ExampleQuery ', 'query ExampleQuery ')
TypeError: GraphQL\Language\Printer::GraphQL\Language\{closure}(): Return value must be of type string, int returned

/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Printer.php:194
/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Visitor.php:277
/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Printer.php:463
/var/www/html/src/vendor/webonyx/graphql-php/src/Language/Printer.php:90
/var/www/html/src/vendor/hyva-themes/magento2-graphql-view-model/src/Model/GraphqlQueryEditor.php:98
/var/www/html/src/vendor/hyva-themes/magento2-graphql-view-model/tests/unit/Model/GraphqlQueryEditorTest.php:119

ERRORS!
Tests: 14, Assertions: 11, Errors: 3.
```

**Comment by schmengler** [comment]
It seems to actually be a bug in this version of `webonyx/graphql-php`: https://github.com/webonyx/graphql-php/blob/v14.11.8/src/Language/Printer.php#L193-L195 (integer nodes are not casted)

**Comment by schmengler** [comment]
I tried to run the tests with the development version of the GraphQL package:
```
composer require webonyx/graphql-php "dev-master as 14.11.9-dev"
```
The error is gone, but unfortunately new ones come because a property has changed from array to object:
```
TypeError: Cannot assign array to property GraphQL\Language\AST\SelectionSetNode::$selections of type GraphQL\Language\AST\NodeList
```

@Vinai not much we can do about it now it seems, but the query editor won't work with Magento 2.4.5 as soon as integer values are involved (e.g. page size)

An update of the module will be necessary as soon as a new version of the graphql package is released, which fixes the issue. I suggest adding version constraints to composer.json now already to not allow installing the current version with a graphql-php version >= 14.12

**Comment by Vinai** [comment]
@schmengler that's a good idea.I really loath the backward incompatible changes in PHP 8.1, but sadly, that's the world we live in.
Would you be willing to open a PR with the suggested change in the version?

**Comment by schmengler** [comment]
of course: https://github.com/hyva-themes/magento2-graphql-view-model/pull/6

**Comment by Vinai** [comment]
Thank you, @schmengler!
Merged and tagged as release 1.0.4
